src/az_ulib_ipc/az_ulib_ipc.c

```diff
@@ -734,101 +732,85 @@ AZ_NODISCARD az_result az_ulib_ipc_split_method_name(
     {
-      // If the next field is a number, it is the package version.
-      if (az_span_atou32(fields[count_field], &package_version_temp) == AZ_OK)
-      {
-        count_field--;
-      }
-      else // If the next field is not a number, set package version as any and use it
-           // as the package name.
-      {
-        package_version_temp = AZ_ULIB_VERSION_DEFAULT;
-      }
+      AZ_ULIB_THROW_IF_ERROR((split_pos == 1), AZ_ERROR_UNEXPECTED_CHAR);
+      package_version_temp = AZ_ULIB_VERSION_DEFAULT;
+    }
+    else
+    {
+      AZ_ULIB_THROW_IF_AZ_ERROR(
+          az_span_atou32(az_span_slice(full_name, 0, split_pos), &package_version_temp));
+    }
+    full_name = az_span_slice_to_end(full_name, split_pos + 1);
 
-      // If package_version was provided, package_name shall not be AZ_SPAN_EMPTY.
-      AZ_ULIB_THROW_IF_ERROR((count_field >= 0), AZ_ERROR_UNEXPECTED_CHAR);
+    // Extract the interface name
+    split_pos = az_span_find(full_name, AZ_SPAN_FROM_STR("."));
+    AZ_ULIB_THROW_IF_ERROR((split_pos > 0), AZ_ERROR_UNEXPECTED_CHAR);
+    interface_name_temp = az_span_slice(full_name, 0, split_pos);
+    full_name = az_span_slice_to_end(full_name, split_pos + 1);
 
-      // Get the package name.
-      *package_name
-          = az_span_create(az_span_ptr(fields[count_field]), az_span_size(fields[count_field]));
-      count_field--;
+    // Extract the interface version
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_span_atou32(full_name, &interface_version_temp));
 
-      if (count_field >= 0) // We do have device name.
-      {
-        *device_name
-            = az_span_create(az_span_ptr(fields[count_field]), az_span_size(fields[count_field]));
-      }
-      else
-      {
-        *device_name = AZ_SPAN_EMPTY;
-      }
+    // Transfer the parser data to the output variables.
+    if (az_span_size(device_name_temp) > 0)
+    {
+      *device_name = az_span_create(az_span_ptr(device_name_temp), az_span_size(device_name_temp));
     }
     else
     {
       *device_name = AZ_SPAN_EMPTY;
-      *package_name = AZ_SPAN_EMPTY;
-      package_version_temp = AZ_ULIB_VERSION_DEFAULT;
     }
-
-    // There is no error, so return the parsed names and versions.
+    *package_name = az_span_create(az_span_ptr(package_name_temp), az_span_size(package_name_temp));
     *package_version = package_version_temp;
-    *interface_name = az_span_create(
-        az_span_ptr(fields[interface_name_field]), az_span_size(fields[interface_name_field]));
+    *interface_name
+        = az_span_create(az_span_ptr(interface_name_temp), az_span_size(interface_name_temp));
     *interface_version = interface_version_temp;
-    if (has_capability_name)
+    if (az_span_size(capability_name_temp) > 0)
     {
-      *capability_name = az_span_create(az_span_ptr(full_name), az_span_size(full_name));
+      *capability_name
+          = az_span_create(az_span_ptr(capability_name_temp), az_span_size(capability_name_temp));
```

Comment:
does this work if `capability_name_temp` is empty?

Reply:
No, that is why we tested in #810.